### PR TITLE
Restrict modification of reservations

### PR DESCRIPTION
**Pull request by shlok007**
Reservation can only be edited by a patron if it satisfies all the given conditions:
- The patron is the reservation requester.
- Status of the reservation is 'Awaiting Review'.
- The modification is made 'x' hours prior to the start of the event as specified by the Nook owner.

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -21,7 +21,17 @@ def show
...
+        flash[:alert] = t('reservations.not_modified')
+        redirect_to reservation_path(@reservation)
+      end
+    else
+      flash[:alert] = t('users.unauthorized')
+      redirect_to root_path
+    end
   end
 
   def update
```
Comment (shubhpatel108) [inline app/controllers/reservations_controller.rb:37]: Check the conditions in update function also. Does this work: Change the id of the reservation in the edit form and try to submit.

Comment (shlok007) [inline app/controllers/reservations_controller.rb:37]: There isn't any field to change the id of the reservation in the edit form.

Comment (shubhpatel108) [inline app/controllers/reservations_controller.rb:37]: Check the action/url attribute of the form using inspect element.

Comment (shlok007) [inline app/controllers/reservations_controller.rb:37]: Yeah, to avoid this condition restriction is to be made in update function too, understood! :+1:

**Pull request merged**